source/common/stats/stats_impl.cc

```diff
@@ -10,6 +10,42 @@
 namespace Envoy {
 namespace Stats {
 
+namespace {
+
+// Round val up to the next multiple of the natural alignment.
+// Note: this implementation only works because 8 is a power of 2.
+size_t roundUpMultipleNaturalAlignment(size_t val) {
+  const size_t multiple = alignof(RawStatData);
+  return (val + multiple - 1) & ~(multiple - 1);
+}
+
+} // namespace
+
+size_t RawStatData::size() {
+  return roundUpMultipleNaturalAlignment(sizeof(RawStatData) + nameSize());
```

Comment:
Can you document the rationale for natural alignment for size? AFAICT, this has to do with how stats are allocated in hot restart shmem, where they are packed together. This isn't obvious when reading the code in the file though.